extensions/common/aws/aws-s3-core/src/main/java/org/eclipse/edc/aws/s3/S3CoreExtension.java

```diff
@@ -75,7 +75,7 @@ public AwsClientProvider awsClientProvider(ServiceExtensionContext context) {
     }
 
     @NotNull
-    private AwsCredentialsProvider createCredentialsProvider(ServiceExtensionContext context) {
+    AwsCredentialsProvider createCredentialsProvider(ServiceExtensionContext context) {
```

Comment:
I would like to make assertion about precedence of EDC configuration over AWS SDK as a guard for future changes. I can not realize the test case after moving the (part of) logic of `S3CoreExtension#createCredentialsProvider` to factory/utils class without passing context (and vault) as argument for clear separation.

I think `S3CoreExtension#createCredentialsProvider` is not purely internal of S3CoreExtensionSince since `AwsCredentialsProvider` is used as argument in many places of AWS API. Making it package private could be reasonable here.

Reply:
context/vault don't need to be passed to an eventual factory beacuse a string supplier could be used, but I agree that this could be overcomplicated for 3 lines of code. I'm ok with this "package-private" thing.